test/androidTest/java/org/thoughtcrime/securesms/mms/VideoCompressionTest.java

```diff
@@ -0,0 +1,61 @@
+package org.thoughtcrime.securesms.mms;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+import android.test.suitebuilder.annotation.LargeTest;
+
+import com.iceteck.silicompressorr.SiliCompressor;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.thoughtcrime.securesms.attachments.Attachment;
+import org.thoughtcrime.securesms.util.MediaUtil;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URISyntaxException;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class VideoCompressionTest {
+    @Test(expected= IOException.class)
+    public void compressVideoTest() throws Exception{
+        //arrange
+        Context testContext = InstrumentationRegistry.getInstrumentation().getContext();
+        InputStream testInput = testContext.getAssets().open("CompressionTestVideo.mp4") ;
+        String outputfolder =  testContext.getCacheDir().toPath().toString();
+        byte[] buffer = new byte[testInput.available()];
+        testInput.read(buffer);
+
+        File targetFile = new File(outputfolder+"/temp.mp4");
+        OutputStream outStream = new FileOutputStream(targetFile);
+        outStream.write(buffer);
+
+        //act
+        String output = SiliCompressor.with(testContext).compressVideo(targetFile.getPath(), outputfolder);
+
+        //assert
+        assert( new File(output).length() < targetFile.length());
+    }
+
+    @Test(expected =  UnsupportedOperationException.class)
+    public void allowedAttachmentTest() throws Exception{
```

Comment:
Same as above

Reply:
It's the proper way of testing for if you want a method to throw an exception.

Reply:
Do we expect an exception in both cases though? The above on seems to assert at the end.

Reply:
Untill someone can figure out how to grant tests read/write permissions, yup.

Reply:
Bryce is right. The test is expecting an exception to be thrown; that's the only way the test will actually pass when run & not throw us an actual error

Reply:
I've made issue #145 to address the inability for android/Instrumented unit tests.

Reply:
I've read over the comments, with the bug issue created I'll approve this.